executor/distsql.go

```diff
@@ -714,75 +704,59 @@ func (e *IndexLookUpExecutor) indexRangesToKVRanges() ([]kv.KeyRange, error) {
 
 // executeTask executes the table look up tasks. We will construct a table reader and send request by handles.
 // Then we hold the returning rows and finish this task.
-func (e *IndexLookUpExecutor) executeTask(task *lookupTableTask, goCtx goctx.Context) {
+func (w *tableWorker) executeTask(goCtx goctx.Context, task *lookupTableTask) {
 	var err error
 	defer func() {
 		task.doneCh <- errors.Trace(err)
 	}()
-
 	var tableReader Executor
-	tableReader, err = e.dataReaderBuilder.buildTableReaderFromHandles(goCtx, &TableReaderExecutor{
-		baseExecutor: newBaseExecutor(e.schema, e.ctx),
-		table:        e.table,
-		tableID:      e.tableID,
-		dagPB:        e.tableRequest,
-	}, task.handles)
+	tableReader, err = w.buildTblReader(goCtx, task.handles)
 	if err != nil {
 		log.Error(err)
```

Comment:
no need to log it since we will write this error back to `task.doneCh`

Reply:
It's is possible that the error is not received.